fbw-common/src/wasm/systems/systems_wasm/src/electrical.rs

```diff
@@ -39,40 +39,59 @@ pub(super) fn electrical_buses<const N: usize>(
 /// function can probably be removed.
 pub(super) fn auxiliary_power_unit(
     is_available_variable: Variable,
-    fuel_valve_number: u8,
-    fuel_pump_number: u8,
+    apu_fuel_valve_number: u8,
+    apu_fuel_pump_number: u8,
 ) -> impl FnOnce(&mut MsfsAspectBuilder) -> Result<(), Box<dyn Error>> {
     move |builder: &mut MsfsAspectBuilder| {
         builder.on_change(
             ExecuteOn::PostTick,
             vec![
                 is_available_variable,
                 Variable::aircraft("APU SWITCH", "Bool", 0),
+                Variable::aircraft("BLEED AIR APU", "Bool", 0),
+                Variable::named("ASU_TURNED_ON"),
+                Variable::named("APU_BLEED_AIR_VALVE_OPEN"),
             ],
             Box::new(move |_, values| {
                 let is_available = to_bool(values[0]);
                 let msfs_apu_is_on = to_bool(values[1]);
+                let msfs_apu_bleed_on = to_bool(values[2]);
+                let asu_turned_on = to_bool(values[3]);
+                let apu_bleed_valve_open = to_bool(values[4]);
 
-                if is_available && !msfs_apu_is_on {
-                    toggle_fuel_valve_and_pump(fuel_valve_number, fuel_pump_number);
+                if (is_available || asu_turned_on) && !msfs_apu_is_on {
+                    set_fuel_valve_and_pump(apu_fuel_valve_number, apu_fuel_pump_number, true);
                     start_apu();
-                } else if !is_available && msfs_apu_is_on {
-                    toggle_fuel_valve_and_pump(fuel_valve_number, fuel_pump_number);
+                } else if !is_available && !asu_turned_on && msfs_apu_is_on {
+                    set_fuel_valve_and_pump(apu_fuel_valve_number, apu_fuel_pump_number, false);
                     stop_apu();
                 }
+
+                if ((is_available && apu_bleed_valve_open) || asu_turned_on) && !msfs_apu_bleed_on {
```

Comment:
The old code also checked for APU_N > 95, is that not needed?

Reply:
If you mean the code in APU.js then no, since is_available already tells us that the APU is ready to supply bleed air.